### DISCUSSION TITLE
Can’t merge QMK with Miryoku

### DISCUSSION THREAD

**Discussion by robrecord** [manna-harbour/miryoku#280] [Q&A] [answered]
My keyboard is new so it only exists in QMK master. I've followed instructions to merge branches locally and failed. I have tried multiple times, deleting my fork each time then following Miryoku instructions to the letter, but I can't get the latest QMK repo to merge with the latest Miryoku branch. It says there's a merge conflict with the readme.org file, and if I add this and commit, I'm stuck with a repo that contains the latest QMK plus only a single file from the Miryoku user directory - no other files are present in this directory. The build process works for the default layout but not Miryoku, because it isn't present. How do I get these branches fully merged? Is there a step missing in the instructions? Please help. Will provide steps taken + logs if required.

Also if I use the build actions in GitHub with merge enabled, the merge fails there too.

EDIT: Merge is broken due to breaking upstream change?
I think this issue might have something to do with the following commit, in November, removing all userspace directories - which when merged deletes miryoku's user directory. https://github.com/qmk/qmk_firmware/pull/22544

EDIT 2: Here's an issue that describes the breaking change: https://github.com/qmk/qmk_firmware/issues/22724 - it's going to be broken for all users unless the instructions are changed.

EDIT 3: I worked around this by doing the following:

```sh
git checkout miryoku
git branch -D miryoku-merge-master  # optionally delete existing merge branch
git checkout -b miryoku-merge-master
git fetch --tags upstream  # may be origin for you
git merge user-keymaps-still-present  # merges are compatible up to this tag
git checkout miryoku .github  # fixes workflow conflicts
git add -A  # fixes readme.md
git commit -m "update qmk master"
make git-submodule
```

**Comment by laurapigeon** [comment]
I've heard this is to do with the [QMK userspace change](https://docs.qmk.fm/#/newbs_external_userspace?id=external-qmk-userspace), and the miryoku docs just aren't up to date with that recent shift. Saying that, I've been trying to use miryoku in userspace and am having trouble with non-constant initialisers so I'm not certain the layout itself is well enough up-to-date.

**Reply by manna-harbour** [reply]
If there are issues with the new feature please make a new Discussion and I'll investigate when I adapt Miryoku QMK.

**Comment by manna-harbour** [comment, marked as answer]
See https://github.com/manna-harbour/miryoku/discussions/287.
